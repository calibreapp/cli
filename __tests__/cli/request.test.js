const {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} = require('../utils')
const site = require('../fixtures/site.json')

describe('request', () => {
  beforeAll(() => setupIntegrationServer(site))
  afterAll(() => teardownIntegrationServer())

  test('missing query argument', async () => {
    const out = await runCLI({ args: 'request', testForError: true })
    expect(out).toMatchSnapshot()
  })

  test('returns response', async () => {
    const slug = 'calibre'
    const out = await runCLI({
      args: `request --query='query GetSite($slug: String!) {organisation{site(slug: $slug){slug}}}' --slug=${slug}`
    })
    expect(out).toMatchSnapshot()
  })
})
