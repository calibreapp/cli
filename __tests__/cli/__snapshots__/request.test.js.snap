// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request missing query argument 1`] = `
"

 Missing required argument: query 

 --- Command documentation
 calibre request

Make a request to the Calibre GraphQL API

Options:

  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --query      GraphQL query to execute                      [string] [required]
  --variables  Pass query variables as named arguments

Examples:
  calibre request --query='query GetSite($slug: String!)
  {organisation{site(slug: $slug){slug}}}' --slug=calibre"
`;

exports[`request returns response 1`] = `
"{
  \\"organisation\\": {
    \\"site\\": {
      \\"slug\\": \\"calibre\\"
    }
  }
}"
`;
